### PR TITLE
Build on windows (mingw) on appveyor

### PR DESCRIPTION
**Pull request by JanSchulz**
For this to work, one of the core commiter should do:

* enable appveyor in the github project (grant access)
* Make an appveyor account and add the github jq repo as a new project

### CODE REVIEW

**Comment by JanSchulz** [comment]
Currently trying to get the build going...

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  #- C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S mingw-w64-i686-toolchain" 
+  - C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S autoconf automake bison flex"
+  # install oniguruma
+  - C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S mingw-w64-i686-oniguruma mingw-w64-x86_64-oniguruma"
+
+
+build_script:
+  - bash -lc "exec 0</dev/null && cd $APPVEYOR_BUILD_FOLDER && autoreconf -i"
+  - bash -lc "exec 0</dev/null && cd $APPVEYOR_BUILD_FOLDER && ./configure --disable-shared --enable-static --enable-all-static"
+  - bash -lc "exec 0</dev/null && cd $APPVEYOR_BUILD_FOLDER && make LDFLAGS=-lshlwapi"
```
Comment (JanSchulz) [inline appveyor.yml]: This is https://github.com/stedolan/jq/issues/911#issuecomment-138118150

**Comment by JanSchulz** [comment]
Logs and history of the builds: https://ci.appveyor.com/project/JanSchulz/jq

This is my repo (aka janschulz/jq), a core committer needs to enable it for this repo...

**Comment by JanSchulz** [comment]
IMO this is ready to merge... Next step would be to enable the tests...

**Comment by JanSchulz** [comment]
Addon with (failing) 32bit builds and (failing) tests: https://github.com/stedolan/jq/pull/1075

**Comment by nicowilliams** [comment]
Thanks!  I'll merge after merging my fixes for the Windows build.

**Comment by nicowilliams** [comment]
I'll merge this in a bit, but enabling appveyor will be something for @stedolan to do.  I'll ping him after I merge.

**Comment by nicowilliams** [comment]
Cherry-picked onto master.  Thanks!

**Comment by nicowilliams** [comment]
One wonders why they couldn't have named the file `.appveyor.yml`...

**Comment by nicowilliams** [comment]
Ah, there's a way to set a custom appveyor.yml file.

**Pull request closed without merging**